### PR TITLE
Split CI workflow jobs

### PR DESCRIPTION
**Pull request by r9-pena**
| | |
|:---|:---|
|**_Originating Project/Creator_**| Ricardo Pena |
|**_Affected Component_**| CI |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Builds Without Errors (y/n)_**|  |
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

Separated the CI test job in the Github workflow to reduce wait time.

## Rationale

CI test time was approximately 14-min, and required all tests to complete before obtaining error logs, if any. By separating the jobs, it is easier to observe test results as they occur and total test time is down to below 9-min.

### CODE REVIEW

File: ci/tests/20-fputil.bash [hunk 1]
```diff
@@ -8,32 +8,63 @@ export SCRIPT_DIR="$(dirname ${BASH_SOURCE})/.."
 . "${SCRIPT_DIR}/helpers.bash"
```
Comment (LeStarch) [inline ci/tests/20-fputil.bash:8]: Can this file be deleted?  It seems to be unused with a preference for other files.

File: .github/actions/entrypoint.sh [hunk 2]
```diff
@@ -4,9 +4,10 @@
 set -e
 set -x
 cd "$GITHUB_WORKSPACE"
-if [ "$GITHUB_WORKFLOW" != "Autodocs" ]
+if "$GITHUB_WORKFLOW" == "Autodocs"
```
Comment (LeStarch) [inline .github/actions/entrypoint.sh:7]: This check should use `[ ]` e.g.:

```
if [ "$GITHUB_WORKFLOW" == "Autodocs" ]
```

File: ci/tests/fputil.bash [hunk 3]
```diff
@@ -81,7 +82,7 @@ function integration_test {
         echo "[INFO] Allowing GDS ${SLEEP_TIME} seconds to start"
         sleep ${SLEEP_TIME}
         # Check the above started successfully
-        ps -p ${GDS_PID} 2> /dev/null 1> /dev/null || fail_and_stop "Failed to run GDS layer headlessly"
```
Comment (LeStarch) [inline ci/tests/fputil.bash:84]: This redirect should be kept, as the error on the right is sufficient.

File: ci/tests/fputil.bash [hunk 4]
```diff
@@ -4,9 +4,10 @@
 #
 # Helpers to test via FP util
 ####
+set -e
 export FPUTIL_TARGETS=("generate" "generate --ut" "build" "build --all" "check --all")
-export FPUTIL_DEPLOYS="${FPRIME_DIR} ${FPRIME_DIR}/Ref ${FPRIME_DIR}/RPI"
-
+# export FPUTIL_DEPLOYS="${FPRIME_DIR} ${FPRIME_DIR}/Ref ${FPRIME_DIR}/RPI"
+export FPUTIL_DEPLOYS="${FPRIME_DIR}/${TEST}"
```
Comment (LeStarch) [inline ci/tests/fputil.bash:10]: In your CI setup, you get the following errors in the Framework test:

 ```
/github/workspace/ci/tests/../tests/fputil.bash: line 28: cd: /github/workspace/Framework/: No such file or directory
```

Perhaps you should just set this variable in each test directly and remove this here?  e.g. in the Framework.bash add a line:

```
export FPUTIL_DEPLOYS="${FPRIME_DIR}"
```

File: ci/tests/fputil.bash [hunk 5]
```diff
@@ -22,24 +23,16 @@ function fputil_action {
     export WORKDIR="${DEPLOYMENT}/${3}"
```
Comment (LeStarch) [inline ci/tests/fputil.bash:23]: You are never passing in `${3}` so this could be:

```
export WORKDIR="${DEPLOYMENT}
```

**Review by LeStarch — CHANGES_REQUESTED** [review]

File: ci/tests/20-fputil.bash (hunk 1 above)
Comment (r9-pena) [reply ci/tests/20-fputil.bash:8]: This is mostly used for fail_and_stop() function. It just provides the errors from the log. If we wanted to move away from logging onto a separate file, we wouldn't need this file.

File: .github/actions/entrypoint.sh (hunk 2 above)
Comment (r9-pena) [reply .github/actions/entrypoint.sh:7]: Incorporated.

File: ci/tests/fputil.bash (hunk 5 above)
Comment (r9-pena) [reply ci/tests/fputil.bash:23]: Incorporated.

File: ci/tests/fputil.bash (hunk 4 above)
Comment (r9-pena) [reply ci/tests/fputil.bash:10]: I replaced {FPUTIL_DEPLOYS} with  {FPRIME_DIR}, within the loop to make it more simple. The errors went away with this change.

File: ci/tests/fputil.bash (hunk 3 above)
Comment (r9-pena) [reply ci/tests/fputil.bash]: Incorporated.

File: ci/tests/fputil.bash (hunk 4 above)
Comment (LeStarch) [reply ci/tests/fputil.bash:10]: @r9-pena why not do something similar to RPI and Ref:

In Ref.bash:
```
fputil_action "${FPRIME_DIR}/Ref" "${target}"
```

In RPI.bash:
```
fputil_action "${FPRIME_DIR}/RPI" "${target}"
```

File: ci/tests/20-fputil.bash (hunk 1 above)
Comment (LeStarch) [reply ci/tests/20-fputil.bash:8]: I don't see where that function is defined in this file?  I am talking about deleting: `ci/tests/20-fputil.bash`

All I see is an unused loop and a bunch of commented-out code.

Comment (r9-pena) [reply ci/tests/20-fputil.bash:8]: Oh yes, I'm sorry. I think I misread the comment. I'll delete it.

File: ci/tests/fputil.bash (hunk 4 above)
Comment (r9-pena) [reply ci/tests/fputil.bash]: Added the following line below, as originally requested, to maintain consistency across all bash files.
```
export FPUTIL_DEPLOYS="${FPRIME_DIR}"
export FPUTIL_DEPLOYS="${FPRIME_DIR}/Ref"
export FPUTIL_DEPLOYS="${FPRIME_DIR}/RPI"
```

**Review by LeStarch — APPROVED** [review]

**Comment by LeStarch** [comment]
I think this is blocked because build will not run, as you removed build.  I will merge with your tests passing

**Pull request merged**